presto-main/src/main/java/com/facebook/presto/operator/scalar/TryFunction.java

```diff
@@ -0,0 +1,177 @@
+    @SqlNullable
+    @SqlType("T")
+    public static Double tryDouble(@SqlType("function(T)") TryDoubleLambda function)
+    {
+        try {
+            return function.apply();
+        }
+        catch (PrestoException e) {
+            return (Double) tryFunctionExceptionHandler(e);
+        }
+    }
+
+    @TypeParameter("T")
+    @TypeParameterSpecialization(name = "T", nativeContainerType = boolean.class)
+    @SqlNullable
+    @SqlType("T")
+    public static Boolean tryBoolean(@SqlType("function(T)") TryBooleanLambda function)
+    {
+        try {
+            return function.apply();
+        }
+        catch (PrestoException e) {
+            return (Boolean) tryFunctionExceptionHandler(e);
+        }
+    }
+
+    @TypeParameter("T")
+    @TypeParameterSpecialization(name = "T", nativeContainerType = Slice.class)
+    @SqlNullable
+    @SqlType("T")
+    public static Slice trySlice(@SqlType("function(T)") TrySliceLambda function)
+    {
+        try {
+            return function.apply();
+        }
+        catch (PrestoException e) {
+            return (Slice) tryFunctionExceptionHandler(e);
+        }
+    }
+
+    @TypeParameter("T")
+    @TypeParameterSpecialization(name = "T", nativeContainerType = Block.class)
+    @SqlNullable
+    @SqlType("T")
+    public static Block tryBlock(@SqlType("function(T)") TryBlockLambda function)
+    {
+        try {
+            return function.apply();
+        }
+        catch (PrestoException e) {
+            return (Block) tryFunctionExceptionHandler(e);
+        }
+    }
+
+    @TypeParameter("T")
+    @TypeParameterSpecialization(name = "T", nativeContainerType = void.class)
+    @SqlNullable
+    @SqlType("T")
+    public static Void tryVoid(@SqlType("function(T)") TryVoidLambda function)
+    {
+        try {
+            return function.apply();
+        }
+        catch (PrestoException e) {
+            return (Void) tryFunctionExceptionHandler(e);
+        }
+    }
+
+    @FunctionalInterface
+    public interface TryLongLambda
+            extends LambdaFunctionInterface
+    {
+        Long apply();
+    }
+
+    @FunctionalInterface
+    public interface TryDoubleLambda
+            extends LambdaFunctionInterface
+    {
+        Double apply();
+    }
+
+    @FunctionalInterface
+    public interface TryBooleanLambda
+            extends LambdaFunctionInterface
+    {
+        Boolean apply();
+    }
+
+    @FunctionalInterface
+    public interface TrySliceLambda
+            extends LambdaFunctionInterface
+    {
+        Slice apply();
+    }
+
+    @FunctionalInterface
+    public interface TryBlockLambda
+            extends LambdaFunctionInterface
+    {
+        Block apply();
+    }
+
+    @FunctionalInterface
+    public interface TryVoidLambda
+            extends LambdaFunctionInterface
+    {
+        Void apply();
+    }
+
+    private static Object tryFunctionExceptionHandler(PrestoException e)
```

Comment:
I suggest to name it `propagateIfUnhandled`, this name is inspired by `Throwables` in guava.